stdlib/runtime/string.gr

```diff
@@ -26,14 +26,28 @@ import NumberUtils from "runtime/numberUtils"
 import { allocateString, allocateArray } from "runtime/dataStructures"
 
 import foreign wasm fd_write: (WasmI32, WasmI32, WasmI32, WasmI32) -> WasmI32 from "wasi_snapshot_preview1"
+import foreign wasm fd_sync : (WasmI32) -> WasmI32 from "wasi_snapshot_preview1"
 
 primitive (!) : Bool -> Bool = "@not"
 primitive (&&) : (Bool, Bool) -> Bool = "@and"
 primitive (||) : (Bool, Bool) -> Bool = "@or"
 
-enum StringList { [], [...](String, StringList) }
+enum StringList { SLEmpty, SLCons(String, StringList) }
 
-let _RUNTIME_TYPE_METADATA_PTR = 0x408n
+@disableGC
+let slConsDisableGc = (a, b) => {
+  Memory.incRef(WasmI32.fromGrain(SLCons))
+  SLCons(a, b)
+}
+
+let mut _RUNTIME_TYPE_METADATA_PTR = 0x01n;
+
+@disableGC
+let initPtr = () => {
+  // hack to avoid incRef on this pointer
```

Comment:
Do we have an issue to allow `disableGC` on `let mut`? If not, can we?

Reply:
#814